snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeTableOperations.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+import java.sql.SQLException;
+import java.util.Map;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.jdbc.UncheckedInterruptedException;
+import org.apache.iceberg.jdbc.UncheckedSQLException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.snowflake.entities.SnowflakeTableMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class SnowflakeTableOperations extends BaseMetastoreTableOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeTableOperations.class);
+  private final String catalogName;
+
+  private final FileIO fileIO;
+  private final TableIdentifier tableIdentifier;
+
+  private final QueryFactory queryFactory;
+
+  private final Map<String, String> catalogProperties;
+
+  protected SnowflakeTableOperations(
+      QueryFactory queryFactory,
+      FileIO fileIO,
+      Map<String, String> properties,
+      String catalogName,
+      TableIdentifier tableIdentifier) {
+    this.queryFactory = queryFactory;
+    this.fileIO = fileIO;
+    this.catalogProperties = properties;
+    this.catalogName = catalogName;
+    this.tableIdentifier = tableIdentifier;
+  }
+
+  @Override
+  public void doRefresh() {
+    String location = null;
+    try {
+      LOG.debug("Getting metadata location for table {}", tableName());
+      location = getTableMetadataLocation();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
```

Comment:
Why are we doing this?

Reply:
Looks like this follows convention of JdbcCatalog: https://github.com/apache/iceberg/blob/0694269cd20a3625d5587d544bec621e9bec305a/core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java#L114

I'm also a bit fuzzy on the actual scenarios where the thread's interrupt bit becomes important when we're going to rethrow an unchecked exception anyways, but this stackoverflow thread seems to provide a good overview. I think the problem comes from the fact that ClientPoolImpl declares throwing the InterruptedException: https://github.com/apache/iceberg/blob/0694269cd20a3625d5587d544bec621e9bec305a/core/src/main/java/org/apache/iceberg/ClientPoolImpl.java#L55

Reply:
Actually, it looks like this is self-inflicted, since we already wrap all the exceptions in the underlying JDBC client wrapper. So this TableOperations class shouldn't need to duplicate the exception wrapping. I'll clean it up.